src/main/java/org/apache/commons/compress/harmony/pack200/MetadataBandGroup.java

```diff
@@ -242,29 +242,40 @@ public void addParameterAnnotation(final int numParams, final int[] annoN, final
         for (Object element : t) {
             final String tag = (String) element;
             T.add(tag);
-            if (tag.equals("B") || tag.equals("C") || tag.equals("I") || tag.equals("S") || tag.equals("Z")) {
-                final Integer value = (Integer) valuesIterator.next();
-                caseI_KI.add(cpBands.getConstant(value));
-            } else if (tag.equals("D")) {
-                final Double value = (Double) valuesIterator.next();
-                caseD_KD.add(cpBands.getConstant(value));
-            } else if (tag.equals("F")) {
-                final Float value = (Float) valuesIterator.next();
-                caseF_KF.add(cpBands.getConstant(value));
-            } else if (tag.equals("J")) {
-                final Long value = (Long) valuesIterator.next();
-                caseJ_KJ.add(cpBands.getConstant(value));
-            } else if (tag.equals("c")) {
-                final String value = (String) valuesIterator.next();
-                casec_RS.add(cpBands.getCPSignature(value));
-            } else if (tag.equals("e")) {
-                final String value = (String) valuesIterator.next();
-                final String value2 = (String) valuesIterator.next();
-                caseet_RS.add(cpBands.getCPSignature(value));
-                caseec_RU.add(cpBands.getCPUtf8(value2));
-            } else if (tag.equals("s")) {
-                final String value = (String) valuesIterator.next();
-                cases_RU.add(cpBands.getCPUtf8(value));
+            switch (tag) {
```

Comment:
Can `tag` be null here?

Reply:
i guess

Reply:
@arturobernalg Would you please rebase on git master?

Reply:
done